From 6d89e871b48cb2a6e5507260a72ac32454eb6875 Mon Sep 17 00:00:00 2001
From: Kazunori Kobayashi <kkobayas@igel.co.jp>
Date: Thu, 11 Jul 2013 14:16:50 +0900
Subject: [PATCH 1/3] ext/dfbvideosink: move meram set up function in order to
 be called regardless of primary or external surface

This fixes meram set up processing not being invoked when an external
surface is set.
---
 ext/directfb/dfbvideosink.c | 16 +++++++---------
 1 file changed, 7 insertions(+), 9 deletions(-)

diff --git a/ext/directfb/dfbvideosink.c b/ext/directfb/dfbvideosink.c
index 49455ca..cd43472 100644
--- a/ext/directfb/dfbvideosink.c
+++ b/ext/directfb/dfbvideosink.c
@@ -2054,17 +2054,15 @@ gst_dfbvideosink_setcaps (GstBaseSink * bsink, GstCaps * caps)
       GST_DEBUG_OBJECT (dfbvideosink, "layer %d now configured to %dx%d %s",
           dfbvideosink->layer_id, lc.width, lc.height,
           gst_dfbvideosink_get_format_name (lc.pixelformat));
-
-#if defined(HAVE_SHMERAM)
-      /*
-       * Meram setup function must be invoked after the pixel format
-       * determination processing.
-       */
-      gst_dfbvideosink_setup_meram (dfbvideosink, caps, video_width,
-          video_height);
-#endif /* defined(HAVE_SHMERAM) */
     }
   }
+#if defined(HAVE_SHMERAM)
+  /*
+   * Meram setup function must be invoked after the pixel format
+   * determination processing.
+   */
+  gst_dfbvideosink_setup_meram (dfbvideosink, caps, video_width, video_height);
+#endif /* defined(HAVE_SHMERAM) */
 #if defined(HAVE_SHVIO)
   /* accept any color format different from destination's one. */
 #else
-- 
1.8.1.2

